subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -1255,6 +1263,304 @@ static int cmd_wifi_ps_wakeup_mode(const struct shell *sh, size_t argc, char *ar
 	return 0;
 }
 
+void parse_mode_args_to_params(const struct shell *sh, int argc,
+			       char *argv[], struct wifi_mode_info *mode,
+			       bool *do_mode_oper)
+{
+	int opt;
+	int option_index = 0;
+
+	static struct option long_options[] = {{"if_index", required_argument, 0, 'i'},
```

Comment:
To me it looks like the interface index should be optional. So if the user does not give it, then we could figure out the index from the default interface. If user gives it, then the network interface (iface variable in various functions), should be resolved using the said if_index variable. WDYT?

Reply:
Let me check this. The code is tested for the interface index 1. I understand what you are stating.